gcovr/gcov_parser.py

```diff
@@ -1024,16 +1051,67 @@ def _find_excluded_ranges(
                 else:  # pragma: no cover
                     pass
 
+            # branche exclusion
+
+            excl_branch_pattern = re.compile(
```

Comment:
Could the two loops be replaced by a function with the regex and lists as argument and returning the excluded ranges? They are doing the same with different variables.

Reply:
...or only the if/elif/else block.

Reply:
I just replaced it all by a function, it's easily changeable now